### PR TITLE
Cancel a download task in notification

### LINKED ISSUE #52: Allow users to stop download from notification
Just a feature request I think this app needs.

### PR DESCRIPTION
**Pull request by JunkFood02**
Also bumped up the version of Kotlin and [Room](https://developer.android.com/jetpack/androidx/releases/room#2.4.0-alpha03), because I use an M1 macOS device to compile(needs Room > 2.4.0-alpha03).
Closes #52

### CODE REVIEW

**Comment by xibr** [comment]
If this is ready, please resolve the conflict.

**Comment by JunkFood02** [comment]
Rebase done 💨

**Comment by xibr** [comment]
I tested it on my phone and some of my attempts to cancel the download didn't work.

**Comment by JunkFood02** [comment]
> I tested it on my phone and some of my attempts to cancel the download didn't work.

Is this can be reproduced or just randomly occurring bugs? I also encountered this bug but a clean re-install solves it, I think it's is the problem of debug variant

File: app/src/main/java/org/yausername/dvd/work/DownloadWorker.kt [hunk 1]
```diff
@@ -113,8 +117,20 @@ class DownloadWorker(appContext: Context, params: WorkerParameters) :
...
+        progress: Int,
+        line: String,
+        tmpFile: File
+    ) {
         val text = line.replace(tmpFile.toString(), "")
+        val intent = Intent(applicationContext, CancelReceiver::class.java)
+            .putExtra("taskId", taskId)
+            .putExtra("notificationId", id)
+
+        val pendingIntent = PendingIntent.getBroadcast(applicationContext, 1, intent, 0)
```
Comment (xibr) [inline app/src/main/java/org/yausername/dvd/work/DownloadWorker.kt:133]:
```suggestion
        val pendingIntent = PendingIntent.getBroadcast(applicationContext, 0, intent, PendingIntent.FLAG_ONE_SHOT)
```

**Review by xibr — COMMENTED** [review]
ًhat about that? 
Like I said in my previous comment but with this change it works fine so far.

**Review by xibr — APPROVED** [review]

**Pull request merged**